src/Data/Mapper/PositionMapper.php

```diff
@@ -27,4 +27,24 @@ public function getPositions($options = [])
 
         return [$this->mapObjects($result), $paginate];
     }
+
+    protected function mapObjects(array $records)
+    {
+        $results = [];
+        foreach ($records as $record) {
+            $entity = new $this->entity;
```

Comment:
The one issue I have with this is that it is so verbose, but like I mentioned before, this is incredibly easy to understand with no magic.